app/steps/step.rb

```diff
@@ -8,7 +8,12 @@ class Step
   class_attribute :attribute_names
 
   def self.step_attributes(*attribute_names)
-    self.attribute_names = attribute_names.map(&:to_s)
-    attr_accessor(*attribute_names)
+    self.attribute_names = attribute_names
+
+    attributes_or_keys = attribute_names.map do |attr|
+      attr.class == Symbol ? attr : attr.keys
```

Comment:
Style: I wonder if we could expand the ternary operator to clarify the branching logic? Probably just me but I often have trouble following it when it's all in the one line.